consensus/core/src/linearizer.rs

```diff
@@ -65,9 +72,16 @@ impl Linearizer {
                         .filter(|ancestor| {
                             // We skip the block if we already committed it or we reached a
                             // round that we already committed.
+                            // TODO: for Fast Path we need to ammend the recursion rule here and allow us to commit blocks all the way up to the `gc_round`.
+                            // Some additional work will be needed to make sure that we keep the uncommitted blocks up to the `gc_round` across commits.
                             !committed.contains(ancestor)
                                 && last_committed_rounds[ancestor.author] < ancestor.round
                         })
+                        .filter(|ancestor| {
+                            // Keep the block if GC is not enabled or it is enabled and the block is above the gc_round. We do this
+                            // to stop the recursion early and avoid going to deep when it's unnecessary.
+                            !gc_enabled || ancestor.round > gc_round
```

Comment:
you don't need to check `gc_enabled` here, right? `gc_round` will be `GENESIS_ROUND` if it is not enabled so it should always return true.

Reply:
Yes that's right. But (and that's something I should have outlined on the PR description as well) I've deliberately decided to use the `gc_enabled` flag on all the paths that we modify the existing behaviour to minimise any chance of introducing bugs from GC to current behaviour. Once GC gets enabled in all our environments then I'll just revisit and cleanup.